website/docs/sdk-reference/dotnet.md

```diff
@@ -240,19 +240,66 @@ User userObject = new User("john@example.com");
 | `Id`       | **REQUIRED.** Unique identifier of a user in your application. Can be any `string` value, even an email address.                |
 | `Email`    | Optional parameter for easier targeting rule definitions.                                                                       |
 | `Country`  | Optional parameter for easier targeting rule definitions.                                                                       |
-| `Custom`   | Optional dictionary for custom attributes of a user for advanced targeting rule definitions. e.g. User role, Subscription type. |
+| `Custom`   | Optional dictionary for custom attributes of a user for advanced targeting rule definitions. E.g. User role, Subscription type. |
 
 ```csharp
 User userObject = new User("#UNIQUE-USER-IDENTIFIER#")
 {
     Email = "john@example.com",
     Country = "United Kingdom",
-    Custom = new Dictionary<string, string> {
-        {"SubscriptionType", "Pro"},
-        {"UserRole", "Admin"}}
+    Custom =
+    {
+        ["SubscriptionType"] = "Pro",
+        ["UserRole"] = "Admin"
+    }
+};
+```
+
+The `Custom` dictionary also allows attribute values other than `string` values:
+
+```csharp
+User userObject = new User("#UNIQUE-USER-IDENTIFIER#")
+{
+    Custom =
+    {
+        ["Rating"] = 4.5,
+        ["RegisteredAt"] = DateTimeOffset.Parse("2023-11-22 12:34:56 +00:00", CultureInfo.InvariantCulture),
+        ["Roles"] = new[] { "Role1", "Role2" }
+    }
 };
 ```
 
+The set of allowed attribute values depends on the comparison type of the condition which references the User Object attribute.
+`string` values are supported by all comparison types (in some cases they need to be provided in a specific format though).
+Some of the comparison types work with other types of values, as described below.
+
+**Text-based comparisons** (EQUALS, IS ONE OF, etc.)
+* accept `string` values,
+* all other values are automatically converted to `string` (a warning will be logged but evaluation will continue as normal).
+
+**SemVer-based comparisons** (IS ONE OF, &lt;, &gt;=, etc.)
+* accept `string` values containing a properly formatted, valid semver value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+**Number-based comparisons** (=, &lt;, &gt;=, etc.)
+* accept `double` values and all other numeric values which can safely be converted to `double`,
+* accept `string` values containing a properly formatted, valid `double` value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+  
+**Date time-based comparisons** (BEFORE / AFTER - available only in the new config model, *Config V2*)
```

Comment:
Not sure if we need to note the V2 availability.

Reply:
Me neither :) I added this because config V2 is not yet available so it might look weird to refer to some non-existent comparators.

Reply:
Yeah, if this will be released earlier, we can remove the note later. Just I'm afraid that no one will remember.

Reply:
Reminder: after discussing this, we decided to remove this part until the related features become available on the dashboard.